CARDS-1835: Update the ui with the new status as the form gets saved

marta- opened the pull request:
[Jira link](https://phenotips.atlassian.net/browse/CARDS-1835)

Quick fix by fetching the form json after successful save to update the flags.

Possible improvements:
* return the status directly in the save response
* add a json processor allowing to fetch only the flags, for less traffic

marta- committed:
CARDS-1835: Update the ui with the new status as the form gets saved

sdumitriu commented on modules/data-entry/src/main/frontend/src/questionnaire/Form.jsx:
```diff
@@ -221,7 +223,13 @@ function Form (props) {
         setLastSaveTimestamp(new Date());
-        onSuccess && onSuccess();
+        if (!disableHeader) {
+          fetchWithReLogin(globalLoginDisplay, `${formURL}.json`)
```
```suggestion
          fetchWithReLogin(globalLoginDisplay, `${formURL}.-dereference.json`)
```

sdumitriu replied:
In theory fetching `${formURL}/statusFlags.json` should get just the flags, but there seems to be a bug in Sling, multi-valued properties are not serialized properly.

marta- committed:
CARDS-1835: Update the ui with the new status as the form gets saved

Load a smaller JSON when updating the flags by not expanding the subject and questionnaire.

Co-authored-by: Sergiu Dumitriu <sdumitriu@users.noreply.github.com>
